services/cms/src/blocks/ExerciseTask/IFrameEditor.tsx

```diff
@@ -29,28 +28,20 @@ const IFrameEditor: React.FC<IFrameEditorProps> = ({ url, props }) => {
       </>
     )
   }
+  let parsedPrivateSpec = null
+  try {
+    parsedPrivateSpec = JSON.parse(props.attributes.private_spec ?? null)
+  } catch (e) {
+    setSpecParseable(false)
+    return
+  }
+
+  console.log(url)
   return (
     <MessageChannelIFrame
       url={url}
-      onCommunicationChannelEstabilished={(port) => {
-        // eslint-disable-next-line i18next/no-literal-string
-        console.info("communication channel established")
-        let parsedPrivateSpec = null
-        try {
-          parsedPrivateSpec = JSON.parse(props.attributes.private_spec ?? null)
-        } catch (e) {
-          setSpecParseable(false)
-          return
-        }
-        const message: SetStateMessage = {
-          // eslint-disable-next-line i18next/no-literal-string
-          message: "set-state",
-          // eslint-disable-next-line i18next/no-literal-string
-          view_type: "exercise-editor",
-          data: parsedPrivateSpec,
-        }
-        port.postMessage(message)
-      }}
+      // eslint-disable-next-line i18next/no-literal-string
+      postThisStateToIFrame={{ view_type: "exercise-editor", data: parsedPrivateSpec }}
```

Comment:
Better pattern without eslint-disable-next-line i18next/no-literal-string is to introduce a constant at the top of the file with upper case name e.g. `const VIEW_TYPE = "exercise-editor"`